core/src/test/java/org/apache/iceberg/TestManifestReaderStats.java

```diff
@@ -149,19 +150,35 @@ private void assertFullStats(DataFile dataFile) {
     Assert.assertNull(dataFile.columnSizes());
     Assert.assertEquals(VALUE_COUNT, dataFile.valueCounts());
     Assert.assertEquals(NULL_VALUE_COUNTS, dataFile.nullValueCounts());
+    Assert.assertEquals(NAN_VALUE_COUNTS, dataFile.nanValueCounts());
     Assert.assertEquals(LOWER_BOUNDS, dataFile.lowerBounds());
     Assert.assertEquals(UPPER_BOUNDS, dataFile.upperBounds());
-    Assert.assertEquals(NAN_VALUE_COUNTS, dataFile.nanValueCounts());
+
+    Assert.assertEquals(FILE_PATH, dataFile.path()); // always select file path in all test cases
   }
 
   private void assertStatsDropped(DataFile dataFile) {
-    Assert.assertEquals(3, dataFile.recordCount()); // always select record count in all test cases
+    Assert.assertEquals(3, dataFile.recordCount()); // record count is not considered as droppable stats
     Assert.assertNull(dataFile.columnSizes());
     Assert.assertNull(dataFile.valueCounts());
     Assert.assertNull(dataFile.nullValueCounts());
+    Assert.assertNull(dataFile.nanValueCounts());
     Assert.assertNull(dataFile.lowerBounds());
     Assert.assertNull(dataFile.upperBounds());
+
+    Assert.assertEquals(FILE_PATH, dataFile.path()); // always select file path in all test cases
+  }
+
+  private void assertNoStats(DataFile dataFile) {
+    Assert.assertEquals(-1L, dataFile.recordCount());
```

Comment:
I think this should always contain the record count, even after `copyWithoutStats`. That's primarily to drop the stats maps, which can be really large.

Reply:
This code actually is not testing results from `copyWithoutStats`, `copyWithoutStats` results are actually tested by another method `assertStatsDropped` which still keeps `recordCount`. In the test case that uses this `assertNoStats`, only `select` but no `filter` operation is applied to the manifest reader, so the reader doesn't project stats columns when reading (since manifest entries don't have to go through evaluators when no filter is applied), and thus only return the field being selected (in this case `file_path`). But I can see that this name is confusing so I updated it a bit to hopefully makes things clearer.